pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -110,6 +118,49 @@ func pushToShifuTelemetryCollectionService(message *http.Response, request *v1al
 	return nil
 }
 
+func getSecret(c *rest.RESTClient, name, ns string) (map[string]string, error) {
+	secret := v1.Secret{}
+	secretPath := fmt.Sprintf("/api/v1/namespaces/%s/%s/%s", ns, SecretResource, name)
+	if err := c.Get().AbsPath(secretPath).Do(context.TODO()).Into(&secret); err != nil {
+		return nil, err
```

Comment:
shall we print the log and then return?

Reply:
The `getSecret` function will return the error, so the error need to be handled by the caller. We might imagine someone calling getSecret() without being sure whether it would log the error so we might be in danger of logging the same errors twice. [ref](https://softwareengineering.stackexchange.com/questions/309038/should-i-log-errors-in-the-function-they-occur-or-pass-them-back-and-log-them-w).